packages/ckeditor5-engine/src/view/matcher.js

```diff
@@ -7,6 +7,9 @@
  * @module engine/view/matcher
  */
 
+import { logWarning } from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
```

Comment:
The `ckeditor5-engine` package is reexported in `ckeditor5` import package. Seems like it would cause circular dependency - that is my assumption, as the precommit hook stops the commit with an error without detailed description.

Reply:
Indeed I see that similar imports in engine are done via your approach - https://github.com/ckeditor/ckeditor5/blob/3b85ab92a2366983b2e29d3a4111c7a7139194a5/packages/ckeditor5-engine/src/conversion/upcasthelpers.js#L10